python/lsst/ts/mtreflector/controller.py

```diff
@@ -0,0 +1,437 @@
+        # addresses for the LSB/MSB
+        if self.channel[:3] == "AIN" or self.channel[:3] == "DAC":
+            addNum *= 2
+        else:
+            # all other DIO is standard 16-bit wide transmission
+            addNum += self.offset_dict[self.channel[:3]]
+        return addNum
+
+    def check_valid(self, value: Any) -> bool:
+        """Wrapper to if value written is valid for labjack channel
+
+        Parameters
+        ----------
+        value : `Any`
+            value to test
+
+        Returns
+        -------
+        valid : `bool`
+            True/False if channel type will accept value
+        """
+        if isinstance(value, MTReflectorStatus):
+            if value == MTReflectorStatus.OPEN or value == MTReflectorStatus.CLOSE:
+                return True
+        return False
+
+    def value(self, state_to_write: MTReflectorStatus) -> bool:
+        """Wrapper to convert OPEN/CLOSE to the actual value to
+            write in case of flipped GPIOs
+
+        Parameters
+        ----------
+        status : `MTReflectorStatus`
+            State to set the reflector to.
+
+        Returns
+        -------
+        state : `bool`
+            Proper True/False to send to labjack
+        """
+        if state_to_write == MTReflectorStatus.OPEN:
+            return True
+        else:
+            return False
+
+
+class MTReflectorController(BaseLabJackDataClient):
+    """Class to handle opening of reflector connected to Labjack Interface
+
+    Parameters
+    ----------
+    config : `types.SimpleNamespace`
+        reflector-specific configuration.
+    log : `logging.Logger` or 'None', optional
+        Logger.
+    simulate : `bool`, optional
+        Run in simulation mode?
+    """
+
+    def __init__(
+        self,
+        config: types.SimpleNamespace,
+        log: logging.Logger | None = None,
+        status: MTReflectorStatus = MTReflectorStatus.UNKNOWN,
+        simulate: bool = False,
+    ) -> None:
+        super().__init__(
+            config=config, topics=config.topics, log=log, simulation_mode=simulate
+        )
+
+        self.config = config
+        self.log = (
+            log.getChild(type(self).__name__)
+            if log is not None
+            else logging.getLogger(type(self).__name__)
+        )
+        reflector_topic = next(
+            topic
+            for topic in self.config.topics
+            if topic["topic_name"] == "reflector_item"
+        )
+
+        self.channel_name = reflector_topic["channel_name"]
+        self.labjack_item = LabjackChannel(channel=self.channel_name, status=status)
+
+        self.log.info(f"Opening reflector on channel {self.channel_name}")
+
+        # Set if connected to the labjack and state data seen,
+        # cleared otherwise.
+        self.status_event = asyncio.Event()
+        self.status_task = utils.make_done_future()
+
+    @classmethod
+    def get_config_schema(cls) -> dict[str, Any]:
+        return yaml.safe_load(
```

Comment:
This schema is an exact copy of the CSC schema. Please eliminate it here and use the one for the CSC. I think you can get rid of the entire method.